### PR TITLE
THCI: recover whitelist and blacklist before reattach after reset

### PR DESCRIPTION
**Pull request by librasungirl**

### CODE REVIEW

File: tools/harness-thci/OpenThread.py [hunk 1]
```diff
@@ -462,9 +464,23 @@ def __startOpenThread(self):
                 else:
                     self.hasActiveDatasetToCommit = False
 
+            # restore whitelist if rejoin after reset
+            if self.isPowerDown and self.isWhiteListEnabled:
+                self.__enableWhiteList()
+                for addr in self.whiteList:
+                    self.addAllowMAC(addr)
```
Comment (xiaom-GitHub) [inline tools/harness-thci/OpenThread.py]: `addAllowMAC` will call `self.whiteList.add(macAddr)` to add all existing entires in whitelist table to whitelist table again. Here, can we replace with `whitelist add` command instead? Same case for blacklist?

File: tools/harness-thci/OpenThread.py [hunk 2]
```diff
@@ -55,6 +55,8 @@ class OpenThread(IThci):
     isPowerDown = False              # indicate if Thread device experiences a power down event
     isWhiteListEnabled = False       # indicate if Thread device enables white list filter
     isBlackListEnabled = False       # indicate if Thread device enables black list filter
+    whiteList = set()                # cache whitelist devices when white list filter is enabled
```
Comment (bukepo) [inline tools/harness-thci/OpenThread.py]: [PEP 8](https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables)
>Use one leading underscore only for non-public methods and instance variables.

Just suggestion: whitelist and blacklist are not phrase.
`whiteList` -> `_whitelist`
`blackList` -> `_blacklist`

File: tools/harness-thci/OpenThread.py (hunk 1 above)
Comment (librasungirl) [inline tools/harness-thci/OpenThread.py]: Here`self.whiteList` is of`set` type which guarantees the uniqueness of elements.
https://docs.python.org/2/library/sets.html

File: tools/harness-thci/OpenThread.py (hunk 2 above)
Comment (librasungirl) [inline tools/harness-thci/OpenThread.py]: will update. Thanks

**Pull request merged**